### ISSUE TITLE
How can I get all the validate data and prediction in each epoch?

### ISSUE THREAD

**Issue by xiaolongwu0713** [skorch-dev/skorch#757]
Here is my code. I am trying to regress xx to yy.
```
net = NeuralNetRegressor(
    model,
    #train_split=predefined_split(valid_set),
    iterator_train__shuffle=True,
    #train_split=5,
    max_epochs=1,
    lr=0.001,
    batch_size=1,
    optimizer=torch.optim.Adam,
    criterion = nn.MSELoss,
    #device='cuda',  # uncomment this to train with CUDA
)
net.fit(xx, yy)
```
It's a regression task, so I want to compare and plot the actual target yy and its predictions after each epoch.
So how could I get the validate data y and the corresponding prediction of y during each epoch? 
I notice there is a history class, however it only log accuracy data not the real raw data.

**Comment by BenjaminBossan** [comment]
Great that you figured something out.

**Edit**: @xiaolongwu0713 I wrote the text below before you closed the issue, it might still be interesting to you.

Hmm, you want to create one plot per epoch? I haven't heard that one before but I guess it's possible. My first idea would be to "hack" the scoring callback by writing a scoring function that actually plots

```python
def my_scoring(net, X_test, y_test):
    # your plotting code here
    return 0  # return a dummy value as the 'score'

from skorch.callbacks import EpochScoring
net = NeuralNetRegressor(..., callbacks=[EpochScoring(my_scoring, name='dummy')])
```

A more proper solution would be to subclass `skorch.callbacks.Callback` and write a proper plotting callback. But the advantage of the "hack" I proposed is that it takes care of stuff you would otherwise have to implement yourself, e.g. providing caching if needed and extracting the right train/valid data.

On top of all of that, you could take a look at our integrations with:

* [neptune](https://skorch.readthedocs.io/en/stable/callbacks.html#skorch.callbacks.NeptuneLogger)
* [sacred](https://skorch.readthedocs.io/en/stable/callbacks.html#skorch.callbacks.SacredLogger)
* [tensorboard](https://skorch.readthedocs.io/en/stable/callbacks.html#skorch.callbacks.TensorBoard)

to help you with better understanding the what the net is doing during training.
